### PR TITLE
Fixes #6477: Create a script to sign files using openssl on unix

### PR DESCRIPTION
**Pull request by peckpeck**
PR replaced by https://github.com/Normation/rudder-packages/pull/644

### CODE REVIEW

File: rudder-agent/SOURCES/signature.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# md4 md5 sha sha1 sha224 sha256 sha384 sha512 whirlpool
+# The oldest openssl we support is 0.9.8 and it supports sha512
+HASH=sha512
+
+# the file to sign
+FILE="$1"
+if [ ! -e "${FILE}" ]
+then
+  echo "Cannot sign: The file ${FILE} doesn't exist"
```
Comment (Kegeruneku) [inline rudder-agent/SOURCES/signature.sh]: Missing an "ERROR: " prefix here

File: rudder-agent/SOURCES/signature.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+# To allow Rudder to provide its own version of openssl
+export PATH=/opt/rudder/bin:$PATH
+
+if openssl -h 2> /dev/null
```
Comment (Kegeruneku) [inline rudder-agent/SOURCES/signature.sh]: "if type openssl" maybe instead ?

**Comment by Kegeruneku** [comment]
I'd rather name this utility signature and not signature.sh :) the suffix is not desirable here nor needed !

File: rudder-agent/SOURCES/signature.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+then
+  true
+else
+  echo "ERROR: openssl binary is missing !"
+  exit 1
+fi
```
Comment (Kegeruneku) [inline rudder-agent/SOURCES/signature.sh]: Why not use:
```
if ! type openssl 2> /dev/null
then
  echo "ERROR: openssl binary is missing !"
  exit 1
fi
```

You'd avoid a useless "true"

File: rudder-agent/SOURCES/signature.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+then
+  echo "Cannot sign: The file ${FILE} doesn't exist"
+  exit 2
+fi
+
+# the key to use for signature
+PRIVKEY=/var/rudder/cfengine-community/ppkeys/localhost.priv
+
+# cfengine  passphrase
+PASSPHRASE="Cfengine passphrase"
```
Comment (Kegeruneku) [inline rudder-agent/SOURCES/signature.sh]: ...
facepalm

**Comment by peckpeck** [comment]
There will be a signature.cmd for windows too, the .sh was here to explicitly distinguish which one will be for wich environment

**Comment by peckpeck** [comment]
PR updated

**Comment by Kegeruneku** [comment]
That's a good reason !

**Comment by Kegeruneku** [comment]
Would you be kind enough to retarget this PR for 3.1 ?

**Comment by Kegeruneku** [comment]
Also this script should be able to try to use Rudder's bundled OpenSSL version if no system one is available (/opt/rudder/bin/openssl)

**Comment by peckpeck** [comment]
PR replaced by https://github.com/Normation/rudder-packages/pull/644

**Pull request closed without merging**